openhands/controller/agent_controller.py

```diff
@@ -351,6 +360,7 @@ async def set_agent_state_to(self, new_state: AgentState) -> None:
                 self.state.iteration is not None
                 and self.state.max_iterations is not None
                 and self._initial_max_iterations is not None
+                and not self.headless_mode
```

Comment:
This looks correct? It's not necessary today, because I think nothing but the UI ever sets the agent in PAUSED state, so scripts don't pass line 353 above.

IMHO we should keep this change. It's more explicit, and it might be useful in the future.